### PR TITLE
[CI]  Update workflow script, ensure functional test determinism, fix coverage tests.

### PR DESCRIPTION
**Pull request by phoenixdong**
### Updates

- Modify test file names and optimize test scripts for a more concise and user-friendly implementation.
- Fix randomness in functional testing to ensure consistent test results.

### Unit Test Instructions

The previous usage has been improved: [https://github.com/FlagOpen/FlagScale/pull/199#issue-2493813612](url).

If you add a new test folder and want it to run in the workflow, rather than just locally, modify the `.github/workflows/test.yml` file:

```yaml
megatron-unit-tests:
  uses: ./.github/workflows/unit-tests.yml
  strategy:
    matrix:
      subset: 
        - data
        - ...
        # - add your own test folder here
```

### Fixed Issues

- Removed broken submodule.
- Online coverage testing cannot recognize code modifications.

### File Removal

- Deleted redundant file: `results_test` in `tests/functional_tests/test_cases/train`.

### CODE REVIEW

File: megatron/tests/functional_tests/local_recipes
Comment (aoyulong) [inline megatron/tests/functional_tests/local_recipes:1]: Why do you delete this folder?

Comment (phoenixdong) [reply megatron/tests/functional_tests/local_recipes:1]: > Why do you delete this folder?

Due to [error](https://github.com/FlagOpen/FlagScale/actions/runs/10765493514/job/29851377995) in github workflow when doing "Checkout Code".

Comment (aoyulong) [reply megatron/tests/functional_tests/local_recipes:1]: Was this folder added by us? We should change Megatron-LM as little as possible.

**Review by aoyulong — APPROVED** [review]
LGTM

**Review by aoyulong — APPROVED** [review]
LGTM

File: megatron/tests/functional_tests/local_recipes
Comment (phoenixdong) [reply megatron/tests/functional_tests/local_recipes:1]: Added during the last merge of megatron code.

Comment (phoenixdong) [reply megatron/tests/functional_tests/local_recipes:1]: > Was this folder added by us? We should change Megatron-LM as little as possible.

Added during the last merge of megatron code.

Comment (phoenixdong) [reply megatron/tests/functional_tests/local_recipes:1]: > > Was this folder added by us? We should change Megatron-LM as little as possible.
> 
> Added during the last merge of megatron code.

Due to the absence of. gitmodules, submodule pointers (i.e. deleted files) are invalid, which can result in an error when checking out the code.

**Pull request merged**